### ISSUE TITLE
Failed to install zls

### ISSUE THREAD

**Issue by thallada** [williamboman/mason.nvim#721]
### I've searched open issues for similar requests

Yes

### I've recently downloaded the latest plugin version of mason.nvim

- [X] Yes

### Problem description

Installation of zls using `:MasonInstall zls` or `:LspInstall zls` fails.

I am using the latest version of mason.nvim and I'm running in Ubuntu WSL (x86_64).

Related issue: https://github.com/williamboman/mason.nvim/issues/618 (this solved a previous 404 error, but now I am getting this separate tar error)

### Expected behavior

That zls is installed successfully.

### Affected packages

zls

### Installation log

```Text
Failed
    ◍ zls
      ▼ Displaying full log
        Downloading file "https://github.com/zigtools/zls/releases/download/0.10.0/x86_64-linux.tar.zst"...
        tar: This does not look like a tar archive
        tar: Skipping to next header
        tar: Exiting with failure status due to previous errors
        spawn: tar failed with exit code 2 and signal 0.
```

### Neovim version (>= 0.7)

NVIM v0.9.0-dev
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3

### Operating system/version

`Linux dynamo-win 4.19.84-microsoft-standard #1 SMP Wed Nov 13 11:44:37 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

### tar version

```
tar (GNU tar) 1.29
Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
```

### Healthcheck

```Text
mason.nvim report ~
- OK neovim version >= 0.7.0
- OK **Go**: `go version go1.19.1 linux/amd64`
- OK **cargo**: `cargo 1.67.0-nightly (eb5d35917 2022-11-17)`
- OK **luarocks**: `/home/linuxbrew/.linuxbrew/bin/luarocks 3.9.1`
- OK **Ruby**: `ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux-gnu]`
- OK **RubyGem**: `2.7.6`
- WARNING **Composer**: not available
- WARNING **PHP**: not available
- OK **npm**: `7.7.6`
- OK **node**: `v15.14.0`
- OK **python3**: `Python 3.6.9`
- OK **pip3**: `pip 9.0.1 from /usr/lib/python3/dist-packages (python 3.6)`
- WARNING **javac**: not available
- OK **java**: `openjdk version "11.0.17" 2022-10-18`
- WARNING **julia**: not available
- OK **wget**: `GNU Wget 1.19.4 built on linux-gnu.`
- OK **curl**: `curl 7.58.0 (x86_64-pc-linux-gnu) libcurl/7.58.0 OpenSSL/1.1.1 zlib/1.2.11 libidn2/2.0.4 libpsl/0.19.1 (+libidn2/2.0.4) nghttp2/1.30.0 librtmp/2.3`
- OK **gzip**: `gzip 1.6`
- OK **tar**: `tar (GNU tar) 1.29`
- WARNING **pwsh**: not available
- OK **bash**: `GNU bash, version 4.4.20(1)-release (x86_64-pc-linux-gnu)`
- OK **sh**: `Ok`
- OK **python3_host_prog**: `Python 3.6.9`
- OK **python3_host_prog pip**: `pip 9.0.1 from /usr/lib/python3/dist-packages (python 3.6)`
- OK GitHub API rate limit. Used: 2. Remaining: 58. Limit: 60. Reset: Sat Dec  3 18:49:08 2022.
```

### Screenshots

![mason](https://user-images.githubusercontent.com/1505923/205466111-8cbccfaa-c0a7-4558-b41d-4095a1aa4a6a.png)

**Comment by thallada** [comment]
So I'm pretty sure this was due to me not having a version of `tar` that supports `zstd`. I was running from Ubuntu 18.04 in WSL. I just upgraded to 22.04 which has this tar version:
```
tar (GNU tar) 1.34
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
```

And, now the `:MasonInstall zls` completes successfully.

Since I don't think this is really a mason.nvim issue, I'll close.

**Comment by williamboman** [comment]
Thanks for reporting back with a solution! `zls` is so far the only package that compresses assets with `zstd`, and I have to say I was a bit surprised by the poor support across platforms when I looked further into it. Seems like `zstd` support was merged to GNU tar on `2018-03-18` and landed in version `1.31`.

**Comment by BVegNow** [comment]
Hi using the latest neovim 0.8.3, tar 1.23 and zstd 1.5.2, and updated mason through packer, on windows 10, i'm stuck at downloading the zls though clicking the link does lead to a downloadable. It can hang there a long time till manually cancelled. The mason.log says this:

[INFO  3/2/2023 3:37:08 pm] ...\packer\opt\mason.nvim/lua/mason-core/installer/init.lua:120: Executing installer for Package(name=zls)
[ERROR 3/2/2023 3:37:38 pm] ...\packer\opt\mason.nvim/lua/mason-core/installer/init.lua:159: Installation failed for Package(name=zls) error="Installation was aborted."

Other packages update fine.

Using the command line I am able to use zstd to unpack the file to a tar archive which does contain the relevant files for zls.
